index.js

```diff
@@ -75,88 +95,126 @@
 //     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
 //   }
 // ]
+
+// function printName (name) {
+//   console.log( `${name}` )
+// }
+
+// users.forEach (printName,val)
+ 
+
 // printNameAndPhones(users)
 // // => "Duane"
-// => "Cell: 555-123-4567"
-// => "Office: 555-456-7890"
-// => "Liza"
-// => "Cell: 555-234-5678"
-// => "Office: 555-567-1234"
+// // => "Cell: 555-123-4567"
+// // => "Office: 555-456-7890"
+// // => "Liza"
+// // => "Cell: 555-234-5678"
+// // => "Office: 555-567-1234"
 // console.log("%c----------", "color: red") 
 
-
 // ***** Callbacks *****
 
-function myMap(array, callback) {
-  const result = []
-  for (let i = 0; i < array.length; i++) {
-    const element = array[i]
-    const value = callback(element)
-    result.push(value)
-  }
-  return result
-}
+// function myMap(array, callback) {
+//   const result = []
+//   for (let i = 0; i < array.length; i++) {
+//     const element = array[i]
+//     const value = callback(element)
+//     result.push(value)
+//   }
+//   return result
+// }
 
 // ***** Callbacks - Question 1 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cCallbacks - Question 1", "color: red")
 
-// console.log(myMap([1,2,3,4,5], triple)) 
+
+console.log("%cCallbacks - Question 1", "color: red")
+
+function myMap (arr,fn) {
+  for (let i = 0; i < arr.length; i++) {
+    fn(arr[i])
+  } 
+}
+
+function triple (number) {
+  console.log(number * 3)
+}
+
+console.log(myMap([1,2,3,4,5], triple)) 
 // // => [3,6,9,12,15]
 
-// console.log(myMap([2,4,6,8], triple))   
+console.log(myMap([2,4,6,8], triple))   
 // // => [6,12,18,24]
-// console.log("%c----------", "color: red") 
-
+console.log("%c----------", "color: red") 
 
 // ***** Callbacks - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cCallbacks - Question 2", "color: red")
 
-// console.log(myMap(["Raffy", "Chase"], greet)) 
-// // => ["Hi, Raffy!", "Hi, Chase!"]
-// console.log("%c----------", "color: red") 
+console.log("%cCallbacks - Question 2", "color: red")
 
+function greet (name) {
+  console.log(`Hi, ${name}!`)
+}
 
+function myMap (arr,fn) {
+  for (let i = 0; i < arr.length; i++) {
+    fn(arr[i])
+  } 
+}
 
-// ***** Scope & Closures *****
+console.log(myMap(["Raffy", "Chase"], greet)) 
```

Comment:
Now we should be able to pass in greet or triple as callbackfunctions to the myMap function, as well as an array, and then based on the callback function, we will alter each element in the array with the callback function.